spacelift/resource_named_webhook.go

```diff
@@ -0,0 +1,202 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/validations"
+)
+
+func resourceNamedWebhook() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_named_webhook` represents a named webhook endpoint used for creating webhooks" +
+			"which are referred to in Notification policies to route messages.",
+
+		CreateContext: resourceNamedWebhookCreate,
+		ReadContext:   resourceNamedWebhookRead,
+		UpdateContext: resourceNamedWebhookUpdate,
+		DeleteContext: resourceNamedWebhookDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "enables or disables sending webhooks. Defaults to `true`.",
+				Optional:    true,
+				Default:     true,
```

Comment:
Please do not set the default for an optional true, we can not check if the attribute was set by the provider (as the default) or through a field. This causes problems when we might want to change this field and the only way to do the change would be a breaking change.